packages/components/src/sandbox/index.native.js

```diff
@@ -0,0 +1,252 @@
+                action: 'resize',
+				width: clientBoundingRect.width,
+				height: clientBoundingRect.height,
+            }));
+		}
+
+		observer = new MutationObserver( sendResize );
+		observer.observe( document.body, {
+			attributes: true,
+			attributeOldValue: false,
+			characterData: true,
+			characterDataOldValue: false,
+			childList: true,
+			subtree: true
+		} );
+
+		window.addEventListener( 'load', sendResize, true );
+
+		// Hack: Remove viewport unit styles, as these are relative
+		// the iframe root and interfere with our mechanism for
+		// determining the unconstrained page bounds.
+		function removeViewportStyles( ruleOrNode ) {
+			if( ruleOrNode.style ) {
+				[ 'width', 'height', 'minHeight', 'maxHeight' ].forEach( function( style ) {
+					if ( /^\\d+(vmin|vmax|vh|vw)$/.test( ruleOrNode.style[ style ] ) ) {
+						ruleOrNode.style[ style ] = '';
+					}
+				} );
+			}
+		}
+
+		Array.prototype.forEach.call( document.querySelectorAll( '[style]' ), removeViewportStyles );
+		Array.prototype.forEach.call( document.styleSheets, function( stylesheet ) {
+			Array.prototype.forEach.call( stylesheet.cssRules || stylesheet.rules, removeViewportStyles );
+		} );
+
+		document.body.style.position = 'absolute';
+		document.body.style.width = '100%';
+		document.body.setAttribute( 'data-resizable-iframe-connected', '' );
+
+		sendResize();
+
+		// Resize events can change the width of elements with 100% width, but we don't
+		// get an DOM mutations for that, so do the resize when the window is resized, too.
+		window.addEventListener( 'resize', sendResize, true );
+} )();`;
+
+const style = `
+	body {
+		margin: 0;
+	}
+	html,
+	body,
+	body > div,
+	body > div iframe {
+		width: 100%;
+	}
+	html.wp-has-aspect-ratio,
+	body.wp-has-aspect-ratio,
+	body.wp-has-aspect-ratio > div,
+	body.wp-has-aspect-ratio > div iframe {
+		height: 100%;
+		overflow: hidden; /* If it has an aspect ratio, it shouldn't scroll. */
+	}
+	body > div > * {
+		margin-top: 0 !important; /* Has to have !important to override inline styles. */
+		margin-bottom: 0 !important;
+	}
+`;
+
+function Sandbox( {
+	html = '',
+	providerUrl = '',
+	scripts = [],
+	styles = [],
+	title = '',
+	type,
+} ) {
+	const ref = useRef();
+	const [ width, setWidth ] = useState( 0 );
+	const [ height, setHeight ] = useState( 0 );
+	const [ contentHtml, setContentHtml ] = useState( getHtmlDoc() );
+
+	const windowSize = Dimensions.get( 'window' );
+	const [ isLandscape, setIsLandscape ] = useState(
+		windowSize.width >= windowSize.height
+	);
+	// On Android, we need to recreate the WebView when the device rotates, otherwise it disappears.
+	// For this purpose, the key value used in the WebView will change when the device orientation gets updated.
+	const key = Platform.select( {
+		android: `${ providerUrl }-${ isLandscape ? 'landscape' : 'portrait' }`,
+		ios: providerUrl,
+	} );
+
+	function getHtmlDoc() {
+		// TODO: Use the device's locale
+		const lang = 'en';
```

Comment:
For now, the embed content will be only in English but we'll use the device's locale to match the content's language with device when tackling [this issue](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3773).

Reply:
This is [the PR](https://github.com/WordPress/gutenberg/pull/33858) that introduces the device's locale to the content.